src/Symfony/Component/Serializer/Normalizer/UidNormalizer.php

```diff
@@ -70,7 +70,13 @@ public function supportsNormalization(mixed $data, string $format = null, array
     public function denormalize(mixed $data, string $type, string $format = null, array $context = []): mixed
     {
         try {
-            return AbstractUid::class !== $type ? $type::fromString($data) : Uuid::fromString($data);
+            if (AbstractUid::class === $type) {
+                trigger_deprecation('symfony/serializer', '6.1', sprintf('Denormalizing to an abstract class in "%s" is deprecated, it will throw an \Error in 7.0.', __CLASS__));
```

Comment:
Rebased.
Removed sprintf.
The \Error won't be caught at all on 7.0 (the catch \Error block will be removed).
I'd prefer to keep the last part of the message because there is no "use XXX instead" alternative, so better be clear that it will throw isn'it?

Reply:
I don't think we want to tell that, because we also don't want ppl to catch the `Error`.
I'd remove. I'd also add a comment in the catch block as a reminder that it should be removed.